services/motion/builtin/builtin_test.go

```diff
@@ -737,59 +737,50 @@ func TestReplanning(t *testing.T) {
 	motionCfg := &motion.MotionConfiguration{PositionPollingFreqHz: 100, ObstaclePollingFreqHz: 1, PlanDeviationMM: epsilonMM}
 
 	type testCase struct {
-		name           string
-		noise          r3.Vector
-		expectedReplan bool
+		name            string
+		noise           r3.Vector
+		expectedSuccess bool
+		expectedErr     string
+		extra           map[string]interface{}
 	}
 
 	testCases := []testCase{
 		{
-			name:           "check we dont replan with a good sensor",
-			noise:          r3.Vector{Y: epsilonMM - 0.1},
-			expectedReplan: false,
+			name:            "check we dont replan with a good sensor",
+			noise:           r3.Vector{Y: epsilonMM - 0.1},
+			expectedSuccess: true,
+			extra:           map[string]interface{}{},
 		},
 		{
-			name:           "check we replan with a noisy sensor",
-			noise:          r3.Vector{Y: epsilonMM + 0.1},
-			expectedReplan: true,
+			// This also checks that `replan` is called under default conditions when "max_replans" is not set
+			name:            "check we fail to replan with a low cost factor",
+			noise:           r3.Vector{Y: epsilonMM + 0.1},
+			expectedErr:     "unable to create a new plan within replanCostFactor from the original",
+			expectedSuccess: false,
+			extra:           map[string]interface{}{"replan_cost_factor": 0.01},
+		},
+		{
+			name:            "check we replan with a noisy sensor",
+			noise:           r3.Vector{Y: epsilonMM + 0.1},
+			expectedErr:     fmt.Sprintf("exceeded maximum number of replans: %d", 4),
+			expectedSuccess: false,
+			extra:           map[string]interface{}{"replan_cost_factor": 10.0, "max_replans": 4},
```

Comment:
In these tests are we not we losing the signal of how many times replanning happened, when it replans & then succeeds?  Any way we could still measure that in these tests, given we expect that will be the happy path?

Reply:
You are correct, the way our testing framework is set up it's difficult to probe this behavior properly.

I'll see what I can set up to get a proper reading on this.

Reply:
I'm good with this being addressed as part of a future PR if you think doing so would be more convenient.

Reply:
In the interest of getting this PR in I would prefer to do that. I'm confident (based on the tests added in `motionplan` plus logging plus reasoning about these tests) that this does in fact do what it says it ought, despite it being difficult to test every specific permutation.

I can add a note on RSDK-3236 to expand these tests as part of that ticket.

Reply:
sounds good, yeah, lets do it in the next ticket